src/ucp/wireup/wireup.c

```diff
@@ -975,6 +967,55 @@ ucp_wireup_get_reachable_mds(ucp_worker_h worker,
     key->reachable_md_map = dst_md_map;
 }
 
+static void
+ucp_wireup_check_config_intersect(ucp_ep_h ep, ucp_ep_config_key_t *new_key)
+{
+    ucp_worker_h worker = ep->worker;
+    uct_ep_h new_uct_eps[UCP_MAX_LANES] = {NULL};
+    ucp_lane_index_t reuse_lane_map[UCP_MAX_LANES] = { UCP_NULL_LANE };
+    ucp_ep_config_key_t *old_key;
+    ucp_lane_index_t lane, reuse_lane;
+
+    if (ep->cfg_index == 0) {
```

Comment:
`UCP_WORKER_CFG_INDEX_NULL`

Reply:
why do we need to skip this config?
wouldn't the for loop below cover this case as well?

Reply:
@yosefe For the ucp_ep in the server side, ```ep->cfg_index``` is initialized to be ```0``` and there's no previous configuration that could be indexed by the ```key```, so the ```old_key->num_lanes``` is ```0``` while the ```new->num_lanes``` is the needed lane number.
I think ```loop``` could cover this case. However, we assert ```old_key->num_lanes >= new_key->num_lanes``` before the loop.

Reply:
> For the ucp_ep in the server side, ep->cfg_index is initialized to be 0
Can you pls point to the code that is doing that?

Reply:
@yosefe Below is the code path.
[ucp_ep_cm_server_create_connected](https://github.com/yosefe/ucx/blob/integration3/src/ucp/wireup/wireup_cm.c#L655)
|--> [ucp_ep_create_to_worker_addr](https://github.com/yosefe/ucx/blob/integration3/src/ucp/core/ucp_ep.c#L346)
     |--> [ucp_ep_new](https://github.com/yosefe/ucx/blob/integration3/src/ucp/core/ucp_ep.c#L87)
            |--> [ucp_ep_config_key_reset(&key)](https://github.com/yosefe/ucx/blob/integration3/src/ucp/core/ucp_ep.c#L103); // key is reset
	    |--> [ucp_worker_get_ep_config](https://github.com/yosefe/ucx/blob/integration3/src/ucp/core/ucp_ep.c#L105)

Reply:
ucp_worker_get_ep_config obtains the next available ep configuration. it is not guaranteed it will always be 0.
so it doesn't make sense to compare ep->cfg_index with anything except UCP_WORKER_CFG_INDEX_NULL

Reply:
@yosefe It use the ```reset key``` to check the indexed configuration. Will it not set ```ep->cfg_index`` to be ```0```?

Reply:
in the current implementation it will, but it's not a right approach to assume it.
we need to check some property of the key (for example, num_lanes==0) rather than the index

Reply:
@yosefe OK. I'll check server ep's ```old_key->num_lanes``` here.